File: packages/dart/lib/src/word/rand_body.dart
import 'dart:math';

import 'package:randino/src/types.dart';
import 'package:randino/src/word/rand_word.dart';

/// The parts of a body, inside and out.
///
/// [randWord] with the theme already chosen; every other parameter is the same.
/// For the detail form, pass [WordTheme.body] to `randWordDetails` — Dart has
/// no overloads, and a second function per theme would be a second wall of them.
///
/// ```dart
/// randBody(language: WordLanguage.ko, count: 3); // [손목, 어깨, 심장]
/// randBody(language: WordLanguage.en, count: 3); // [Wrist, Shoulder, Heart]
/// ```
List<String> randBody({
  WordLanguage? language,
  int count = 1,
  RandRealism realism = RandRealism.real,
  RandVocabulary vocabulary = RandVocabulary.full,
  int? minLength,
  int? maxLength,
  String? startsWith,
  bool unique = false,

  /// Where the randomness comes from: `Random.secure()` for a value nobody may
  /// predict, `Random(42)` for one that has to come out the same every run.
  Random? random,
}) => randWord(
  language: language,
  theme: WordTheme.body,
  count: count,
  realism: realism,
  vocabulary: vocabulary,
  minLength: minLength,
  maxLength: maxLength,
  startsWith: startsWith,
  unique: unique,
  random: random,
);

File: packages/javascript/lib/word/randBody.ts
import type { RandThemedWordOptions, WordDetail } from '../_types/global.js';
import { themedWord } from './wordGenerator.js';

/**
 * The parts of a body, inside and out.
 *
 * `randWord({ theme: 'body' })` with the theme already chosen; every other
 * option is the same.
 *
 * @example
 * randBody({ language: 'ko', count: 3 }); // ['손목', '어깨', '심장']
 * randBody({ language: 'en', count: 3 }); // ['Wrist', 'Shoulder', 'Heart']
 */
export function randBody(options?: RandThemedWordOptions & { output?: 'value' }): string[];
/**
 * Body words with the language and theme each one came from.
 *
 * @example
 * randBody({ language: 'ko', output: 'detail' });
 * // [{ word: '손목', language: 'ko', theme: 'body' }]
 */
export function randBody(options: RandThemedWordOptions & { output: 'detail' }): WordDetail[];
export function randBody(options: RandThemedWordOptions = {}): string[] | WordDetail[] {
	return themedWord('body', options);
}
